lib/spack/spack/architecture.py

```diff
@@ -184,8 +184,23 @@ def __contains__(self, cpu_flag):
         return cpu_flag in self.microarchitecture
 
     def optimization_flags(self, compiler):
+        try:
+            # Try to check if the current compiler comes with a
+            # version number supported by archspec.cpu
+            compiler_version = compiler.version
+            v, _, suffix = str(compiler_version).partition('-')
+            _ = tuple(int(y) for y in v.split('.'))
```

Comment:
I am wondering if this is better factored out in a function in `archspec.cpu` like:
```python
version_number, suffix = archspec.cpu.version_components(version)
```
using a regex like:
```python
regex = r'([\d.]*)(-?)(.*)'
```
for the match. If so here we could write more cleanly:
```
version_number, suffix = archspec.cpu.version_components(version)
if version_number is None:
    # Assume custom version and do the right thing
```

Reply:
why does this need an assignment (i.e. why does it need `_ = ...`)?  I'm suprised that wasn't caught by `flake8`, as you typically only use `_` with *multiple* assignment to omit a value.  Can't you just use the tuple constructor?

I don't really have a preference w.r.t. the regex -- I would probably do it with a regex but it looks ok the way it is.

Reply:
Outdated by 3bd0d52